solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -717,26 +718,35 @@ contract RandomBeacon is Ownable {
 
     /// @notice Creates a new relay entry.
     /// @param entry Group BLS signature over the previous entry.
-    function submitRelayEntry(bytes calldata entry) external {
+    /// @param groupMembers Group member ids that participated in dkg (excluding IA/DQ).
+    function submitRelayEntry(
+        bytes calldata entry,
+        uint32[] calldata groupMembers
+    ) external {
         uint256 currentRequestId = relay.currentRequestID;
 
         Groups.Group storage group = groups.getGroup(
             relay.currentRequestGroupID
         );
 
+        require(
+            group.membersHash == keccak256(abi.encode(groupMembers)),
+            "Invalid group members"
+        );
+
         uint256 slashingAmount = relay.submitEntry(entry, group.groupPubKey);
 
         if (slashingAmount > 0) {
-            address[] memory groupMembers = sortitionPool.getIDOperators(
-                group.members
+            address[] memory activeOperators = sortitionPool.getIDOperators(
```

Comment:
Why `activeOperators` and not just `groupMemberIds`? `groupMembers` does not necessarily contain only active group members.

Reply:
[This](https://github.com/keep-network/keep-core/pull/2768#discussion_r773098164) comment
> So the list selected by the sortition pool (1) could be: selectedMembers or selectedOperators or operators, especially that in the Sortition Pool we use a term operator for the parties.
2.a. Could be activeOperators
2.b. Could be misbehavedOperators.

Ok, I thought they are active. Since we're calling sortition pool then maybe we can name them `operators`?

Reply:
We're getting IDs for `groupMembers`, so naming it `groupMemberIds` makes sense to me.

Reply:
☝🏻 they are not selected by the sortition pool; the sortition pool is just keeping track of IDs

Reply:
Actually, aren't we converting IDs to addresses here? Shouldn't the variable be called `groupMembersAddresses`?

Reply:
e538ed6